securityscanutils/trivy_scanner_test.go

```diff
@@ -0,0 +1,109 @@
+package securityscanutils_test
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	"github.com/rotisserie/eris"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/solo-io/go-utils/osutils/executils"
+	. "github.com/solo-io/go-utils/securityscanutils"
+)
+
+var _ = Describe("Trivy Scanner", func() {
+	var (
+		t                         *TrivyScanner
+		inputImage                string
+		inputMarkdownTemplateFile string
+		outputDir, outputFile     string
+		err                       error
+	)
+
+	JustBeforeEach(func() {
+		t = NewTrivyScanner(executils.CombinedOutputWithStatus)
+		inputMarkdownTemplateFile, err = GetTemplateFile(MarkdownTrivyTemplate)
+		Expect(err).NotTo(HaveOccurred())
+		outputDir, err := ioutil.TempDir("", "")
+		Expect(err).NotTo(HaveOccurred())
+		outputFile = filepath.Join(outputDir, "test_report.docgen")
+		inputImage = "quay.io/solo-io/gloo:1.11.1"
```

Comment:
It would be ideal to write tests that don't depend on the ability to reach our Docker repo as that is liable to flake
Mocking the command executor would be one solution

Reply:
Fixed! I would assume the benchmarking tests have to hit the Docker repo though right?

Reply:
I think this is a good point to raise and I see a couple of potential ideas around this:
1. I think it would be valuable to unit test the behavior by mocking the cmdExecutor so we're not hitting trivy directly (this was the intention of making it injectable)
2. I think it would also be valuable to have an integration test that does hit trivy and docker, since our method is designed to be resilient to flakes, since we don't want those to cause the job to fail. If the test is flaky that to me indicates that the job would be similarly flakey (which we want to avoid)

RE bencharking: I guess we don't need to benchmark this test specifically, but I do think we want to ensure that we are retrying at the proper times and not retrying at other times

Reply:
That works out really well! The tests as they stand now are all run with mock executors so we can test our logic independent of trivy and we multipurpose our benchmarking test as an integration test as well since it has to hit trivy.